contrib/win32/win32compat/pwd.c

```diff
@@ -262,6 +262,32 @@ w32_getpwuid(uid_t uid)
 	return ret;
 }
 
+PSID
+getusid(void)
+{
+	LPWSTR user_sid = NULL;
+	HANDLE token = 0;
+	DWORD info_len = 0;
+	TOKEN_USER* info = NULL;
+	PSID ret = NULL;
+
+	errno = 0;
+
+	if (OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &token) == FALSE ||
+		GetTokenInformation(token, TokenUser, NULL, 0, &info_len) == TRUE ||
```

Comment:
We need to split these conditions out since not all mean a errno = ENOMEM. Also, in failure cases, we need to log a debug() or a debug2() message (for easy diagnosis).